src/server/zset_family.cc

```diff
@@ -3028,16 +3079,19 @@ void ZSetFamily::GeoRadiusByMember(CmdArgList args, ConnectionContext* cntx) {
   string_view key = ArgS(args, 0);
   // member to latlong, set shape.xy
   string_view member = ArgS(args, 1);
+  OpResult<double> member_score;
   auto cb = [&](Transaction* t, EngineShard* shard) {
-    return OpScore(t->GetOpArgs(shard), key, member);
+    member_score = OpScore(t->GetOpArgs(shard), key, member);
+    return OpStatus::OK;
   };
-  OpResult<double> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
-  if (result.status() == OpStatus::WRONG_TYPE) {
+  cntx->transaction->Schedule();
+  cntx->transaction->Execute(std::move(cb), false);
+  if (member_score.status() == OpStatus::WRONG_TYPE) {
```

Comment:
you can not call `Execute(..)` and then exit early. In fact, please a test that runs this command on a wrong type and see the testing framework fails because of this early return. 

If you want to return early you should call `transaction->Conclude();`